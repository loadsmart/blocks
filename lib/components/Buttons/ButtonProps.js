"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ButtonDisplayState;
(function (ButtonDisplayState) {
    ButtonDisplayState[ButtonDisplayState["Normal"] = 0] = "Normal";
    ButtonDisplayState[ButtonDisplayState["Disabled"] = 1] = "Disabled";
    ButtonDisplayState[ButtonDisplayState["Loading"] = 2] = "Loading";
})(ButtonDisplayState = exports.ButtonDisplayState || (exports.ButtonDisplayState = {}));
var ButtonHeight;
(function (ButtonHeight) {
    ButtonHeight[ButtonHeight["Regular"] = 50] = "Regular";
    ButtonHeight[ButtonHeight["Short"] = 42] = "Short";
})(ButtonHeight = exports.ButtonHeight || (exports.ButtonHeight = {}));
