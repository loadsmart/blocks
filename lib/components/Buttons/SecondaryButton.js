"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var ButtonProps_1 = require("./ButtonProps");
var react_native_1 = require("react-native");
var res_1 = require("../../res/");
var SecondaryButton = /** @class */ (function (_super) {
    __extends(SecondaryButton, _super);
    function SecondaryButton() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.renderNormalState = function () {
            var styles = stylesFromProps(_this.props);
            return (react_1.default.createElement(react_1.default.Fragment, null,
                _this.props.icon && (react_1.default.createElement(react_native_1.Image, { source: _this.props.icon, style: [styles.icon, _this.props.iconStyle] })),
                react_1.default.createElement(react_native_1.Text, { style: styles.text }, _this.props.title.toUpperCase())));
        };
        _this.renderLoadingState = function () {
            var styles = stylesFromProps(_this.props);
            return react_1.default.createElement(react_native_1.ActivityIndicator, { color: res_1.Colors.Charcoal, style: styles.activityIndicator });
        };
        _this.onPress = function () {
            var _a = _this.props, displayState = _a.displayState, onPress = _a.onPress;
            if (displayState === ButtonProps_1.ButtonDisplayState.Normal && onPress) {
                onPress();
            }
        };
        return _this;
    }
    SecondaryButton.prototype.render = function () {
        var styles = stylesFromProps(this.props);
        return (react_1.default.createElement(react_native_1.TouchableOpacity, { onPress: this.onPress.bind(this), style: [styles.wrapper, this.props.style] },
            this.props.displayState !== ButtonProps_1.ButtonDisplayState.Loading && this.renderNormalState(),
            this.props.displayState === ButtonProps_1.ButtonDisplayState.Loading && this.renderLoadingState()));
    };
    SecondaryButton.defaultProps = {
        displayState: ButtonProps_1.ButtonDisplayState.Normal,
    };
    return SecondaryButton;
}(react_1.PureComponent));
exports.default = SecondaryButton;
var stylesFromProps = function (props) {
    return react_native_1.StyleSheet.create({
        wrapper: {
            flexDirection: 'row',
            justifyContent: 'center',
            alignItems: 'center',
            borderWidth: 1,
            borderColor: res_1.Colors.Charcoal,
            borderRadius: props.buttonHeight ? props.buttonHeight / 2.0 : ButtonProps_1.ButtonHeight.Regular / 2.0,
            paddingHorizontal: 8,
            height: props.buttonHeight ? props.buttonHeight : ButtonProps_1.ButtonHeight.Regular,
        },
        icon: {
            marginRight: 4,
        },
        text: {
            fontFamily: res_1.Fonts.SharpSansBold,
            fontSize: 12,
            marginLeft: 4,
            color: res_1.Colors.Charcoal,
        },
        activityIndicator: {
            alignSelf: 'center',
            position: 'absolute',
        },
    });
};
