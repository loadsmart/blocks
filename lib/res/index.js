"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var colors_1 = __importDefault(require("./colors"));
exports.Colors = colors_1.default;
var fonts_1 = __importDefault(require("./fonts"));
exports.Fonts = fonts_1.default;
var images_1 = __importDefault(require("./images"));
exports.Images = images_1.default;
