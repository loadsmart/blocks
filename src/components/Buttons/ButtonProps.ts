import { ImageSourcePropType, ImageStyle, StyleProp, ViewProps } from 'react-native'

export enum ButtonDisplayState {
  Normal,
  Disabled,
  Loading,
}

export enum ButtonHeight {
  Regular = 50,
  Short = 42,
}

export enum ButtonDisplayStyle {
  Dark,
  Light,
}

export interface ButtonProps extends ViewProps {
  icon?: ImageSourcePropType
  iconStyle?: StyleProp<ImageStyle>
  title: string
  buttonHeight?: ButtonHeight
  displayState?: ButtonDisplayState
  displayStyle?: ButtonDisplayStyle
  onPress?: () => void
}
