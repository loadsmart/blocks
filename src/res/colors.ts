const Colors = {
  AlgaeGreen: '#14d64d',
  BrownGreyThree: '#9b9b9b',
  Charcoal: '#3c3d3a',
  CoralPink: '#fd576a',
  DarkGrey: '#666666',
  OrangePeel: '#ff9800',
  LightGray: '#d4d4d4',
  Tuna: '#484a4c',
  VeryLightGray: '#eee',
  White: '#fff',
  Ghost: '#c4ccd6',
  Transparent: '#00000000',
}

export default Colors
