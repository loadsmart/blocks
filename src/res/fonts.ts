const Fonts = {
  SharpSansBold: 'SharpSans-Bold',
  SharpSansItalic: 'SharpSans-BoldItalic',
  SharpSansBook: 'SharpSans-Book',
  SharpSansBookItalic: 'SharpSans-BookItalic',
  SharpSansExtrabold: 'SharpSans-Extrabold',
  SharpSansExtraboldItalic: 'SharpSans-ExtraboldItalic',
  SharpSansLight: 'SharpSans-Light',
  SharpSansLightItalic: 'SharpSans-LightItalic',
  SharpSansMedium: 'SharpSans-Medium',
  SharpSansMediumItalic: 'SharpSans-MediumItalic',
  SharpSansSemibold: 'SharpSans-Semibold',
  SharpSansSemiboldItalic: 'SharpSans-SemiboldItalic',
  SharpSansThin: 'SharpSans-Thin',
  SharpSansThinItalic: 'SharpSans-ThinItalic',
}

export default Fonts
