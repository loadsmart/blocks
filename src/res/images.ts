const Images = {
  Chevron: require('../../assets/images/chevron.png'),
  PlusSign: require(`../../assets/images/plus-sign.png`),
  SmallCheck: require(`../../assets/images/small-check.png`),
  Warning: require(`../../assets/images/warning.png`),
  PasswordShow: require('../../assets/images/password-show.png'),
  PasswordHide: require('../../assets/images/password-hide.png'),
}

export default Images
