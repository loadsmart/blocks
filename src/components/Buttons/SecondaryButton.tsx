import React, { PureComponent } from 'react'
import { ActivityIndicator, Image, StyleSheet, Text, TouchableOpacity } from 'react-native'
import { Colors, Fonts } from '../../res/'
import { ButtonDisplayState, ButtonHeight, ButtonProps, ButtonDisplayStyle } from './ButtonProps'

export default class SecondaryButton extends PureComponent<ButtonProps> {
  public static defaultProps = {
    displayState: ButtonDisplayState.Normal,
  }

  public render() {
    const styles = stylesFromProps(this.props)
    return (
      <TouchableOpacity
        onPress={this.onPress.bind(this)}
        style={[styles.wrapper, this.props.style]}
      >
        {this.props.displayState !== ButtonDisplayState.Loading && this.renderNormalState()}
        {this.props.displayState === ButtonDisplayState.Loading && this.renderLoadingState()}
      </TouchableOpacity>
    )
  }

  private renderNormalState = () => {
    const styles = stylesFromProps(this.props)
    return (
      <>
        {this.props.icon && (
          <Image source={this.props.icon} style={[styles.icon, this.props.iconStyle]} />
        )}
        <Text style={styles.text}>{this.props.title.toUpperCase()}</Text>
      </>
    )
  }

  private renderLoadingState = () => {
    const styles = stylesFromProps(this.props)
    return <ActivityIndicator color={Colors.Charcoal} style={styles.activityIndicator} />
  }

  private onPress = () => {
    const { displayState, onPress } = this.props
    if (displayState === ButtonDisplayState.Normal && onPress) {
      onPress()
    }
  }
}

const stylesFromProps = (props: ButtonProps) =>
  StyleSheet.create({
    wrapper: {
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      borderWidth: 1,
      borderColor: props.displayStyle === ButtonDisplayStyle.Light ? Colors.White : Colors.Charcoal,
      borderRadius: props.buttonHeight ? props.buttonHeight / 2.0 : ButtonHeight.Regular / 2.0,
      paddingHorizontal: 8,
      height: props.buttonHeight ? props.buttonHeight : ButtonHeight.Regular,
    },
    icon: {
      marginRight: 4,
    },
    text: {
      fontFamily: Fonts.SharpSansBold,
      fontSize: 12,
      marginLeft: 4,
      color: props.displayStyle === ButtonDisplayStyle.Light ? Colors.White : Colors.Charcoal,
    },
    activityIndicator: {
      alignSelf: 'center',
      position: 'absolute',
    },
  })
