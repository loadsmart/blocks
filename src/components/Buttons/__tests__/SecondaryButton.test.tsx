import React from 'react'
import { ActivityIndicator, Text, TouchableOpacity, Image } from 'react-native'
import renderer from 'react-test-renderer'
import SecondaryButton from '../SecondaryButton'
import { ButtonDisplayState, ButtonProps, ButtonHeight, ButtonDisplayStyle } from '../ButtonProps'
import { Images } from '../../../res/'

describe('SecondaryButton', () => {
  let testRenderer: renderer.ReactTestRenderer
  let onPressCallback: jest.Mock

  beforeEach(() => {
    onPressCallback = jest.fn()
  })

  describe('when state is normal', () => {
    beforeEach(() => {
      const props: ButtonProps = {
        icon: Images.PlusSign,
        title: 'accept',
        displayState: ButtonDisplayState.Normal,
        onPress: onPressCallback,
        buttonHeight: ButtonHeight.Regular,
      }

      testRenderer = renderer.create(<SecondaryButton {...props} />)
    })

    it('renders icon', () => {
      const image = testRenderer.root.findByType(Image)
      expect(image.props.source).toBe(Images.PlusSign)
    })

    it('renders title', () => {
      const title = testRenderer.root.findByType(Text)
      expect(title.props.children).toBe('ACCEPT')
    })

    it('does not render spinner', () => {
      expect(testRenderer.root.findAllByType(ActivityIndicator)).toHaveLength(0)
    })

    it('will execute onPress callback', () => {
      const touchable = testRenderer.root.findByType(TouchableOpacity)
      touchable.props.onPress()
      expect(onPressCallback).toHaveBeenCalledTimes(1)
    })

    it('has a valid snapshot', () => {
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })
  })

  describe('when state is loading', () => {
    beforeEach(() => {
      const props: ButtonProps = {
        icon: Images.PlusSign,
        title: 'accept',
        displayState: ButtonDisplayState.Loading,
        onPress: onPressCallback,
        buttonHeight: ButtonHeight.Regular,
      }
      testRenderer = renderer.create(<SecondaryButton {...props} />)
    })

    it('does not render icon', () => {
      expect(testRenderer.root.findAllByType(Image)).toHaveLength(0)
    })

    it('does not render title', () => {
      expect(testRenderer.root.findAllByType(Text)).toHaveLength(0)
    })

    it('renders spinner', () => {
      expect(testRenderer.root.findAllByType(ActivityIndicator)).toHaveLength(1)
    })

    it("won't execute onPress callback", () => {
      const touchable = testRenderer.root.findByType(TouchableOpacity)
      touchable.props.onPress()
      expect(onPressCallback).not.toHaveBeenCalled()
    })

    it('has a valid snapshot', () => {
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })
  })

  describe('when state is disabled', () => {
    beforeEach(() => {
      const props: ButtonProps = {
        title: 'accept',
        displayState: ButtonDisplayState.Disabled,
        onPress: onPressCallback,
        buttonHeight: ButtonHeight.Regular,
      }
      testRenderer = renderer.create(<SecondaryButton {...props} />)
    })

    it('renders title', () => {
      const title = testRenderer.root.findByType(Text)
      expect(title.props.children).toBe('ACCEPT')
    })

    it('does not render spinner', () => {
      expect(testRenderer.root.findAllByType(ActivityIndicator)).toHaveLength(0)
    })

    it("won't execute onPress callback", () => {
      const touchable = testRenderer.root.findByType(TouchableOpacity)
      touchable.props.onPress()
      expect(onPressCallback).not.toHaveBeenCalled()
    })

    it('has a valid snapshot', () => {
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })
  })

  describe('height', () => {
    it('has snapshot for type normal', () => {
      testRenderer = renderer.create(
        <SecondaryButton title={'test'} buttonHeight={ButtonHeight.Regular} />
      )
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it('has snapshot for type short', () => {
      testRenderer = renderer.create(
        <SecondaryButton title={'test'} buttonHeight={ButtonHeight.Short} />
      )
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })

    it('has snapshot for type undefined', () => {
      testRenderer = renderer.create(<SecondaryButton title={'test'} />)
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })
  })

  it('renders for style dark', () => {
    testRenderer = renderer.create(
      <SecondaryButton title={'Cancel'} displayStyle={ButtonDisplayStyle.Dark} />
    )

    expect(testRenderer.toJSON).toMatchSnapshot()
  })

  it('renders for style light', () => {
    testRenderer = renderer.create(
      <SecondaryButton title={'Cancel'} displayStyle={ButtonDisplayStyle.Light} />
    )

    expect(testRenderer.toJSON).toMatchSnapshot()
  })
})
