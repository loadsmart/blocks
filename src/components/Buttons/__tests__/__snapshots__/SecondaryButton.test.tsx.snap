// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryButton height has snapshot for type normal 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#3c3d3a",
      "borderRadius": 25,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#3c3d3a",
        "fontFamily": "SharpSans-Bold",
        "fontSize": 12,
        "marginLeft": 4,
      }
    }
  >
    TEST
  </Text>
</View>
`;

exports[`SecondaryButton height has snapshot for type short 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#3c3d3a",
      "borderRadius": 21,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 42,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#3c3d3a",
        "fontFamily": "SharpSans-Bold",
        "fontSize": 12,
        "marginLeft": 4,
      }
    }
  >
    TEST
  </Text>
</View>
`;

exports[`SecondaryButton height has snapshot for type undefined 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#3c3d3a",
      "borderRadius": 25,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#3c3d3a",
        "fontFamily": "SharpSans-Bold",
        "fontSize": 12,
        "marginLeft": 4,
      }
    }
  >
    TEST
  </Text>
</View>
`;

exports[`SecondaryButton renders for style dark 1`] = `[Function]`;

exports[`SecondaryButton renders for style light 1`] = `[Function]`;

exports[`SecondaryButton when state is disabled has a valid snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#3c3d3a",
      "borderRadius": 25,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#3c3d3a",
        "fontFamily": "SharpSans-Bold",
        "fontSize": 12,
        "marginLeft": 4,
      }
    }
  >
    ACCEPT
  </Text>
</View>
`;

exports[`SecondaryButton when state is loading has a valid snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#3c3d3a",
      "borderRadius": 25,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#3c3d3a"
    hidesWhenStopped={true}
    size="small"
    style={
      Object {
        "alignSelf": "center",
        "position": "absolute",
      }
    }
  />
</View>
`;

exports[`SecondaryButton when state is normal has a valid snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#3c3d3a",
      "borderRadius": 25,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/plus-sign.png",
      }
    }
    style={
      Array [
        Object {
          "marginRight": 4,
        },
        undefined,
      ]
    }
  />
  <Text
    style={
      Object {
        "color": "#3c3d3a",
        "fontFamily": "SharpSans-Bold",
        "fontSize": 12,
        "marginLeft": 4,
      }
    }
  >
    ACCEPT
  </Text>
</View>
`;
